src/components/editor/metaform-editor-right-drawer.tsx

```diff
@@ -147,7 +194,7 @@ const MetaformEditorRightDrawer: React.FC<Props> = ({
               label={ strings.draftEditorScreen.editor.features.fieldTitle }
               value={ field.title ? field.title : "" }
               onChange={ event => updateFormField({
-                ...field, title: event.target.value, name: `${sectionIndex}.${fieldIndex}`
+                ...field, title: event.target.value, name: `${pendingForm!.sections![sectionIndex].title}.${event.target.value}`.replace(/ /g, "-")
```

Comment:
Would look cleaner if value passed to _name_ was constructed into a variable earlier and then that variable was passed to _name_. Also updateFormFIeld() arguments should be splitted into its own lines.